### DISCUSSION TITLE
export to text in specific format

### DISCUSSION THREAD

**Discussion by noideawhatshappening** [hluk/CopyQ#2901] [Q&A] [answered]
Hi. I've been pulling my hair on this one: I'm trying to create the following:

"Shortcut1" copies the text in a tab called "notes", in an item that has the tag "title". This item is emptied before copying something new.
"Shortcut2" copies in the same tab "notes", in an item tagged as "text".  This is also emptied right before I add something new.
"Shortcut3", exports to a text file. filename is the contents of item "title". contents of the file is the contents of item "text".

Anything that points me in the right direction helps. I read MOST documentation and script examples and I still can't pull this one. Please help.

**Comment by hluk** [comment, marked as answer]
> "Shortcut1" copies the text in a tab called "notes", in an item that has the tag "title". This item is emptied before copying something new.

I hope the following command help ([here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them)):

```ini
[Command]
Command="
    copyq:
    tab('notes');

    // Remove \"title\" tagged item
    const sel = ItemSelection().select(/title/, plugins.itemtags.mimeTags);
    sel.removeAll();

    // Send Ctrl+C to the active window to copy
    copy();
    const text = clipboard();
    let newItem = {};
    newItem[mimeText] = text;
    newItem[plugins.itemtags.mimeTags] = 'title';
    add(newItem);"
Icon=
IsGlobalShortcut=true
Name=Shortcut 1
```

> "Shortcut3", exports to a text file. filename is the contents of item "title". contents of the file is the contents of item "text".

You can write to a file with the following code:

```js
// NOTE: Use / as file separator in paths.
let file = File(filePath);
if (!f.openWriteOnly()) {
    throw 'Failed to open the file: ' + f.errorString();
}
if (f.write(dataToWrite) == -1) {
    throw 'Failed to save the file: ' + f.errorString();
}
```

**Reply by noideawhatshappening** [reply]
Wow, this looks a lot more complicated than I thought it would be. Here was me browsing documentation and other's people issues , trying to whip up something coherent with some higher level code. I would have never pulled this off.  The guru (did not know it was AI) had very different, but similar ideas to Chat-GPT. var this, var that, select, select, that kind of thing. Thank you very muchly!

**Reply by noideawhatshappening** [reply]
I managed to make the the first 2 shortcuts work. however, the export does not work.   "Error: execve: Permission denied". tried running copyq as sudo, changed permissions of the path I am saving this to and still same error.
